engine/consensus/matching/engine.go

```diff
@@ -207,29 +206,22 @@ func (e *Engine) process(originID flow.Identifier, event interface{}) error {
 
 // onReceipt processes a new execution receipt.
 func (e *Engine) onReceipt(originID flow.Identifier, receipt *flow.ExecutionReceipt) error {
+	initialState, finalState, err := receiptIntegrityCheck(receipt)
+	if err != nil {
+		return engine.NewInvalidInputErrorf("%w", err)
+	}
+
 	log := e.log.With().
 		Hex("origin_id", originID[:]).
 		Hex("receipt_id", logging.Entity(receipt)).
 		Hex("result_id", logging.Entity(receipt.ExecutionResult)).
-		Hex("previous_id", receipt.ExecutionResult.PreviousResultID[:]).
+		Hex("previous_result", receipt.ExecutionResult.PreviousResultID[:]).
 		Hex("block_id", receipt.ExecutionResult.BlockID[:]).
 		Hex("executor_id", receipt.ExecutorID[:]).
+		Hex("initial_state", initialState).
+		Hex("final_state", finalState).
 		Logger()
 
-	resultFinalState, ok := receipt.ExecutionResult.FinalStateCommitment()
-	if !ok { // discard receipt
-		log.Error().Msg("execution receipt without FinalStateCommit received")
-		return engine.NewInvalidInputErrorf("execution receipt without FinalStateCommit: %x", receipt.ID())
-	}
-	log = log.With().Hex("final_state", resultFinalState).Logger()
-
-	resultInitialState, ok := receipt.ExecutionResult.InitialStateCommit()
-	if !ok { // discard receipt
-		log.Error().Msg("execution receipt without InitialStateCommit received")
-		return engine.NewInvalidInputErrorf("execution receipt without InitialStateCommit: %x", receipt.ID())
-	}
-	log = log.With().Hex("initial_state", resultInitialState).Logger()
-
```

Comment:
⚠️ 
with this revision, I feel we loose significant amount of logging information in case of problems. Assume we receive a receipt without initial or final state. The previous version would log a message including all the following details about the receipt which has problems: 
https://github.com/onflow/flow-go/blob/77e19c1c20d820031eed64ac0be3e39e1877060c/engine/consensus/matching/engine.go#L211-L216

With the new code version, we will only know which Execution Node _forwarded_ us the faulty receipt (but not any more details, like the identity of the EN who originally produced the receipt `executor_id`, what block the receipt was for `executor_id`, etc).  Also, note that we can't retrieve this information from the consensus node, as id doesn't persist faulty receipts.

Reply:
good point, I'm going to revert some of the changes, and maybe improve the logging and error handling in a separate PR.